layersvt/api_dump.h

```diff
@@ -241,9 +242,69 @@ class ConditionalFrameOutput {
     }
 };
 
+#ifdef __ANDROID__
+
+class AndroidLogcatBuf : public std::streambuf {
+   public:
+    static constexpr size_t bufsize = 0x400;  // ... or some other suitable buffer size
+    AndroidLogcatBuf() { this->setp(buffer, buffer + bufsize - 1); }
+
+   private:
+    virtual int overflow(int c) override {
+        if (c == traits_type::eof()) {
+            return traits_type::not_eof(c);
+        }
+        buffer[bufsize - 1] = traits_type::to_char_type(c);
+        flush_rest();
+        buffer[bufsize - 1] = '\0';
+        return traits_type::not_eof(c);
+    }
+
+    virtual int sync() override {
+        flush_rest();
+        if (!previous_rest.empty()) {
+            __android_log_print(4, "VALIDATION", "%s", previous_rest.c_str());
```

Comment:
* We probably don't want to use the magic number 4
* We may want to use the tag name defined as a macro somewhere in the repo.